bai-tap-buoi-11/main.js

```diff
@@ -0,0 +1,127 @@
+// Bài 1:
+// Cách 1:
+const getRepeatArray = (array, number) => {
+  for (let i = 0; i < number; i++) {
+    console.log(array)
+  }
+}
+
+// Cách 2:
+const getrepeatArrayVer2 = (array, number) => {
+  let index = 0
+  while (index < number) {
+    console.log(array)
+    index++
+  }
+}
+
+// Bài 2:
+const getReverseArray = array => {
+  let reverseArray = []
+  let j = 0
+  for (let i = array.length - 1; i >= 0; i--) {
+    reverseArray[j++] = array[i] 
+  }
+  console.log(reverseArray)
+}
+
+// Bài 3:
+const removeValue = array => {
+  const valuesToRemove = [null, undefined, false, '', 0]
+  const newArray = array.filter(itemArray => !valuesToRemove.includes(itemArray))
+  console.log(newArray)
+}
+
+// Bài 4:
+const creatObject = (array = []) => {
+  const newObject = {}
+  array.forEach(item => {
+    newObject[item [0]] = item [1]
+  })
+  return newObject
+}
+
+// Bài 5:
+const sortArray = array => {
+  for (let i = 0; i <= array.length; i++) {
+    for (let j = i + 1; j < array.length; j++) {
+      if (array[i] > array[j]) {
+        let swap = 0
+        swap = array[i]
+        array[i] = array[j]
+        array[j] = swap
+      }
+    }
+  }
+  return array
+}
+
+// Bài 6:
+const checkObject = object => {
+  typeof object === 'object' ? true : false
```

Comment:
sai
hàm trên ko return về gì cả

Hơn nữa nếu lệnh trên đúng thì viết như thế này là đủ:

```suggestion
  return typeof object === 'object'
```


Không đi viết các câu lệnh dạng
```javascript
 return true ? true : false
```
hoặc
```javascript
  if (true) {
    return true
  } else {
    return false
  }
```
Chỉ cần return về biểu thức điều kiện thôi

Và hàm trên cũng đang sai, 1 là vì ko return về cái gì cả, 2 là vd nếu đã có return thì cũng sai: truyền array vào cũng trả về true